src/RestAPI/dlwsrestapi.py

```diff
@@ -1050,9 +1053,17 @@ def post(self):
                 for i in range(nums[role]):
                     pod_names.append(job_id + "-" + role + str(i))
 
-        # endpoints should be in ["ssh", "ipython"]
-        if any(elem not in ["ssh", "ipython"] for elem in requested_endpoints):
-            return ("Bad request, endpoints only allowed in [\"ssh\", \"ipython\"]: %s" % requested_endpoints), 400
+        interactive_ports = []
+        # endpoints should be ["ssh", "ipython", "tensorboard", {"name": "port name", "podPort": "port on pod in 40000-49999"}]
+        for interactive_port in [ elem for elem in requested_endpoints if elem not in ["ssh", "ipython", "tensorboard"] ]:
+            if any(required_field not in interactive_port for required_field in ["name", "podPort"]):
+                # if ["name", "port"] not in interactive_port:
+                return ("Bad request, interactive port should have \"name\" and \"podPort\"]: %s" % requested_endpoints), 400
+            if int(interactive_port["podPort"]) < 40000 or int(interactive_port["podPort"]) > 49999:
+                return ("Bad request, interactive podPort should in range 40000-49999: %s" % requested_endpoints), 400
+            if len(interactive_port["name"]) > 16:
```

Comment:
Why do we have this restriction? Where is this "name" coming from?

Reply:
Comes from that k8s have a length limitation on 'id'.